addon/-private/system/store.js

```diff
@@ -958,13 +951,15 @@ Store = Service.extend({
     ```
 
     @method getReference
-    @param {String} type
+    @param {String} modelName
     @param {String|Integer} id
     @since 2.5.0
     @return {RecordReference}
   */
-  getReference: function(type, id) {
-    return this._internalModelForId(type, id).recordReference;
+  getReference(modelName, id) {
+    modelName = this._classKeyFor(modelName);
```

Comment:
Agreed, hadn't thought about that, will refactor to use a new variable name. I'd also like to bring up (as a separate PR) that it would be nice to officially deprecate support for camelCase modelName usage.